evalml/automl/automl_search.py

```diff
@@ -644,7 +644,7 @@ def _compute_cv_scores(self, pipeline, X, y):
                 y_threshold_tuning = None
                 if self.optimize_thresholds and self.objective.problem_type == ProblemTypes.BINARY and self.objective.can_optimize_threshold:
                     X_train, X_threshold_tuning, y_train, y_threshold_tuning = train_test_split(X_train, y_train, test_size=0.2, random_state=self.random_state)
-                cv_pipeline = pipeline.clone()
+                cv_pipeline = pipeline.clone(pipeline.random_state)
```

Comment:
Yep

RE your comment in standup this week: @eccabay and I chose this behavior because we felt copying random_state under the hood was dirty. However, in light of our recent conversation about #1324 (component/pipeline determinism), perhaps we should revisit this, and have `clone` copy `random_state` by default. Do you have thoughts @freddyaboulton @eccabay ? We can move the discussion to an issue and resolve there.

Reply:
Great idea! I filed #1340 .